metals/src/main/scala/scala/meta/internal/metals/codeactions/ExtractValueCodeAction.scala

```diff
@@ -0,0 +1,311 @@
+      startBlockPos.setEnd(statStart.getStart())
+
+      val noIndentation = equalsPos.startLine == defn.body.pos.startLine
+
+      // Scala 3 optional braces
+      if (canUseBracelessSyntax(stat, source)) {
+        // we need to create a new indented region
+        if (noIndentation) {
+          Seq(
+            new l.TextEdit(
+              startBlockPos,
+              s"""|
+                  |$additionalIndent$valueString
+                  |$innerIndentation""".stripMargin
+            )
+          )
+        }
+        // make sure existing indentation after `=` is correct
+        else {
+          val statIndentation =
+            indentationLength(source, stat.pos.start - stat.pos.startColumn)
+          val statAdditionalIndentation =
+            if (statIndentation <= defnLineIndentation.size)
+              (defnLineIndentation.size - statIndentation) + additionalIndent.size
+            else 0
+          val indentStat = blank.stringRepeat(statAdditionalIndentation)
+
+          Seq(
+            new l.TextEdit(
+              statStart,
+              s"""|$indentStat${valueString.trim()}
+                  |$innerIndentation""".stripMargin
+            )
+          )
+        }
+        // Scala 2
+      } else {
+        val startBlockText =
+          // we should indent the stat
+          if (noIndentation)
+            s"""| {
+                |$additionalIndent$valueString
+                |$innerIndentation""".stripMargin
+          // stat should alredy be indented correctly
+          else
+            s"""| {
+                |$valueString
+                |$innerIndentation""".stripMargin
+
+        val startBlockEdit =
+          new l.TextEdit(startBlockPos, startBlockText)
+        val endBracePos = defn.pos.toLSP
+        endBracePos.setStart(endBracePos.getEnd())
+        val endBraceEdit =
+          new l.TextEdit(endBracePos, s"\n$defnLineIndentation}")
+        Seq(startBlockEdit, endBraceEdit)
+      }
+    }
+
+    // no braces are needed
+    edits.getOrElse {
+      val range = stat.pos.toLSP
+      val start = range.getStart()
+      start.setCharacter(0)
+      range.setEnd(start)
+      Seq(new l.TextEdit(range, s"$valueString\n"))
+    }
+
+  }
+
+  /**
+   * `(` is contained between function and it's arguments
+   */
+  private def applyHasParens(apply: Term.Apply) = {
+    apply.tokens.exists { t =>
+      t.is[Token.LeftParen] &&
+      t.pos.start >= apply.fun.pos.end &&
+      apply.args.headOption.forall(_.pos.start >= t.pos.end)
+    }
+  }
+
+  private def indentationLength(text: String, lineStart: Int): Int = {
+    var i = lineStart
+    while (i < text.length() && (text(i) == '\t' || text(i) == ' ')) {
+      i += 1
+    }
+    i - lineStart
+  }
+
+  /**
+   * Check if it's possible to use braceless syntax and whther
+   * it's the preffered style in the file.
+   */
+  private def canUseBracelessSyntax(stat: Tree, source: String) = {
```

Comment:
☹️ there is a decent amount of code in here just trying to get the braceless syntax right. I honestly wish we didn't have to do this and just inserted code with braces if needed, and then let formatting figure it out.

Reply:
I know.. this actually the worst part to figure out and I spent way too much time on it :S